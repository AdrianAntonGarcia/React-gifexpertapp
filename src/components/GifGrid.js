import React, { Fragment } from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

const GifGrid = ({ category }) => {
	const { data: images, loading } = useFetchGifs(category);

	// console.log(data);
	console.log(loading);

	// /**
	//  * Con el use effect ejecutamos la función dependiendo de la condición del segundo parámetro
	//  */

	/**
	 * Si lo ponemos aquí cada vez que react detecte un cambio en cualquier estado
	 * recargará el componente ejecutando la función cada vez
	 */
	// getGifs();
	return (
		<Fragment>
			<h3 className="animate__animated animate__fadeIn">{category}</h3>
			{loading && <p className="animate__animated animate__flash">Loading</p>}
			<div className="card-grid">
				{images.map((img) => (
					<GifGridItem key={img.id} {...img} />
				))}
			</div>
		</Fragment>
	);
};

export default GifGrid;
