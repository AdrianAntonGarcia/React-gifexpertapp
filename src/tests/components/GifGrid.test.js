import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
/**
 * Cuando se use esta importación en los componentes de nuestra aplicación,
 * va a fingir con lo que nosotros definamos más abajo
 */
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas del componente <GifGrid/>", () => {
	const category = "One Punch";

	// beforeEach(() => {
	// 	// jest.resetAllMocks();
	// 	wrapper = shallow(<GifGrid category={category} />);
	// });

	test("Debe de comprobar que el componente cargue correctamente", () => {
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true,
		});
		const wrapper = shallow(<GifGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
	});

	test("Debe de mostrar items cuando se cargan imagenes con el useFetchGifs", () => {
		/**
		 * Para esto usamos los mocks
		 */
		const gifs = [
			{
				id: "ABC",
				url: "https://localhost/cualquier/cosa.jpg",
				title: "Cualquier cosa",
			},
			{
				id: "123",
				url: "https://localhost/cualquier/cosa.jpg",
				title: "Cualquier cosa",
			},
		];

		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false,
		});
		const wrapper = shallow(<GifGrid category={category} />);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find("p").exists()).toBe(false);
		/**
		 * Poniendo solamente el nombre del componente ya lo seleccionamos
		 */
		expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
	});
});
