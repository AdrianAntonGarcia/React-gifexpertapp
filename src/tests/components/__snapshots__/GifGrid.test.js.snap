// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifGrid/> Debe de comprobar que el componente cargue correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas del componente <GifGrid/> Debe de mostrar items cuando se cargan imagenes con el useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
